src/redux/explorer.ts

```diff
@@ -0,0 +1,1054 @@
+      transactions_limit: TRANSACTIONS_LIMIT,
+    },
+    scope: [
+      `${Network.arbitrum}-transactions`,
+      `${Network.polygon}-transactions`,
+    ],
+  },
+];
+
+/**
+ * Configures a chart retrieval request for assets.
+ *
+ * @param assetCodes The asset addresses.
+ * @param currency The currency to use.
+ * @param chartType The `ChartType` to use.
+ * @param action The request action.
+ * @returns Arguments for an `emit` function call.
+ */
+const chartsRetrieval = (
+  assetCodes: string[],
+  currency: string,
+  chartType: ChartType,
+  action: SocketGetActionType = 'get'
+): SocketEmitArguments => [
+  action,
+  {
+    payload: {
+      asset_codes: assetCodes,
+      charts_type: chartType,
+      currency: toLower(currency),
+    },
+    scope: ['charts'],
+  },
+];
+
+/**
+ * Emits an asset price request. The result is handled by a listener in
+ * `listenOnAssetMessages`.
+ *
+ * @param assetAddress The address to fetch.
+ */
+export const fetchAssetPrices = (assetAddress: string) => (
+  _: Dispatch,
+  getState: AppGetState
+) => {
+  const { assetsSocket } = getState().explorer;
+  const { nativeCurrency } = getState().settings;
+
+  const payload: SocketEmitArguments = [
+    'get',
+    {
+      payload: {
+        asset_codes: [assetAddress],
+        currency: toLower(nativeCurrency),
+      },
+      scope: ['prices'],
+    },
+  ];
+  assetsSocket?.emit(...payload);
+};
+
+/**
+ * Unsubscribes from existing asset subscriptions.
+ */
+const explorerUnsubscribe = () => (_: Dispatch, getState: AppGetState) => {
+  const {
+    addressSocket,
+    addressSubscribed,
+    assetsSocket,
+  } = getState().explorer;
+  const { nativeCurrency } = getState().settings;
+  const { pairs } = getState().uniswap;
+  if (!isNil(addressSocket)) {
+    addressSocket.emit(
+      ...addressSubscription(addressSubscribed!, nativeCurrency, 'unsubscribe')
+    );
+    addressSocket.close();
+  }
+  if (!isNil(assetsSocket)) {
+    assetsSocket.emit(
+      ...assetPricesSubscription(keys(pairs), nativeCurrency, 'unsubscribe')
+    );
+    assetsSocket.close();
+  }
+};
+
+/**
+ * Disables the fallback explorer if it has been enabled.
+ */
+const disableFallbackIfNeeded = () => (
+  dispatch: ThunkDispatch<AppState, unknown, ExplorerDisableFallbackAction>,
+  getState: AppGetState
+) => {
+  const { fallback, assetsTimeoutHandler } = getState().explorer;
+
+  if (fallback) {
+    logger.log('😬 Disabling fallback data provider!');
+    dispatch(fallbackExplorerClearState());
+  }
+  assetsTimeoutHandler && clearTimeout(assetsTimeoutHandler);
+
+  dispatch({
+    type: EXPLORER_DISABLE_FALLBACK,
+  });
+};
+
+/**
+ * Checks whether or not a response message from Zerion is valid.
+ *
+ * @param msg The Zerion asset message.
+ * @returns Whether or not the response is valid.
+ */
+const isValidAssetsResponseFromZerion = (msg: AddressAssetsReceivedMessage) => {
+  // Check that the payload meta is valid
+  if (msg?.meta?.status === 'ok') {
```

Comment:
Since you're touching all the files while this is being used (pretty much all the explores + redux/data),
Can you move the string to a constant? maybe something like `DISPERSION_SUCCESS_CODE`